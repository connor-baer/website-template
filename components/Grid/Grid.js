import styled from '@emotion/styled';

import { pageWidthStyles, gridStyles } from '../../styles/shared';

const Grid = styled('div')(pageWidthStyles, gridStyles);

export default Grid;
