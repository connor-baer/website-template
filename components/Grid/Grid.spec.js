import React from 'react';

import Grid from './Grid';

describe('Grid', () => {
  function renderGrid(fn, props = {}) {
    return fn(<Grid {...props} />);
  }

  it('should render with default styles', () => {
    const { container } = renderGrid(render);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const html = renderGrid(renderToHtml);
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
