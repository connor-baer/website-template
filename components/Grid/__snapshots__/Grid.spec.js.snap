// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render with default styles 1`] = `
.circuit-0 {
  max-width: 67.5rem;
  margin-right: auto;
  margin-left: auto;
  padding-right: 1rem;
  padding-left: 1rem;
  display: grid;
  grid-template-columns: repeat(12,[col-start] minmax(0,1fr));
  grid-column-gap: 1rem;
}

@media (min-width:600px) {
  .circuit-0 {
    padding-right: 2rem;
    padding-left: 2rem;
  }
}

<div
  class="circuit-0 circuit-1"
/>
`;
